### PR TITLE
Hey what if you could build and deconstruct light switches.

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

Makes it so you can build and deconstruct light switches.
Also resprites them to be a bit larger; and use overlays instead of five separate icon states

## Why It's Good For The Game

Why can't we build light switches again?

## Changelog

:cl:
add: Building lightswitches.
/:cl:

### CODE REVIEW

File: code/_onclick/item_attack.dm [hunk 1]
```diff
@@ -10,7 +10,7 @@
 /obj/item/proc/melee_attack_chain(mob/user, atom/target, params)
 	var/is_right_clicking = LAZYACCESS(params2list(params), RIGHT_CLICK)
 
-	if(tool_behaviour && (target.tool_act(user, src, tool_behaviour, is_right_clicking) & TOOL_ACT_MELEE_CHAIN_BLOCKING))
+	if(tool_behaviour && target.tool_act(user, src, tool_behaviour, is_right_clicking))
```
Comment (ZephyrTFA) [inline code/_onclick/item_attack.dm:13]: This was done to correct unintended behaviour, returning any truthy value from tool_act is supposed to short circuit the melee chain

**Review by francinum — CHANGES_REQUESTED** [review]
Being flat out, these sprites are... not good. The light switch is huge and I don't... Really see what the problem is with having the entire sprite change instead of being built from overlays.

besides the fact the former is a bit less flexible.

File: code/_onclick/item_attack.dm (hunk 1 above)
Comment (Kapu1178) [reply code/_onclick/item_attack.dm:13]: This does not appear to be unintended. According to the comment for TOOL_ACT_MELEE_CHAIN_BLOCKING, this appears to be the desired behavior.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/_onclick/item_attack.dm (hunk 1 above)
Comment (ZephyrTFA) [reply code/_onclick/item_attack.dm:13]:
```suggestion
	if(tool_behaviour && (target.tool_act(user, src, tool_behaviour, is_right_clicking) & TOOL_ACT_MELEE_CHAIN_BLOCKING))
```

**Review by Kapu1178 — APPROVED** [review]

**Review by Kapu1178 — APPROVED** [review]

**Pull request merged**